consul.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+var request = require('request')
+
+var SWARM_PREFIX = 'swarm/docker/swarm/nodes/'
+
+function Consul () {
+  this.CONSUL_HOST = process.env.CONSUL_HOST
+  if (!this.CONSUL_HOST) {
+    throw new Error('CONSUL_HOST must be defined')
+  }
+}
+
+Consul.prototype._makeRequest = function (url, cb) {
+  request.get(url, {}, function (err, res, body) {
+    if (err) { return cb(err) }
+    body = JSON.parse(body)
+    body = body.map(function (v) {
```

Comment:
Need an `Array.isArray` check before running map on the body. It is possible that the JSON is valid and not an array.